2-promises/solution.js

```diff
@@ -24,16 +24,46 @@ const id = yourRandomMethod() //third run
 7. log the resultant fullname, or the error, at the end
 */
 
+const validateNames = require('./firstnames');
+const validateLastnames = require('./lastnames');
+
+
 function solution() {
-    // YOUR SOLUTION GOES HERE
+  // YOUR SOLUTION GOES HERE
+  
+  // You generate your id value here
+  const randomNumber = () =>  new Promise ( resolve => {
+    const number = Math.floor(Math.random() * 100 * (Date.now() % 2 === 0 ? 1 : -1));
+    resolve(number);
+  })
 
-    // You generate your id value here
+  let firstname;
+  let lastname;
 
+  randomNumber()
+  .then( id => {
     // You call the lastnames method with your id
-
+    return validateLastnames(id);
+  })
+  .then ( lastnameResult => {
     // You call the firstname method
-
-    // You log the fullname, or error, here
+    lastname = lastnameResult;
+    return validateNames(lastname);
+  })
+  .then( firstnameResult => {
+    firstname = firstnameResult;
```

Comment:
Can you return a promise here with the firstName so you don't have to assign it but instead pass it to the finally?